Clean flare comp

hush-hush opened the pull request:
### What does this PR do?

This PR is better reviewed commit by commit:
-   Adding params to the flare comp
   
    The flare component is no longer a passthrough to the pkg/flare but
    holds all the logic to create and send flares.
   
    The API for the component has been cleaned and params are now given at
    initialization instead.
   
    The logic to send flare is, for now, in the helpers package to be used
    by binaries still relying on pkg/flare.
-   Initializing flare component through FX
   
    This removes all the duplicate logic about default log files.
    We are also now using the flare comp to generate flares from the GUI,
    CLI, agent API and systray.
-   Migrate code out of the flare pkg into comp or the CLI
   
    Only the CLI can generate the pprof profiles. Moving the logic around
    them to the CLI and improving testing. This remove dependencies on
    pkg/flare and simplify the flare logic.
-   Use new flare builder in pkg/flare/archive

### Describe how to test/QA your changes

Try to create a flare from:
- CLI locally
- CLI with a runing agent
- systray
- GUI
- from the cluster-agent
- from the security-agent

For each flare check that the content match a flare generated from a previous version of the agent.

Side note: the wording content of the `local` file, for CLI flare, has change but should behave in the same way.

clamoriniere reviewed (approved):
looks good for @DataDog/container-integrations files

hush-hush committed:
Adding params to the flare comp

The flare component is no longer a passthrough to the pkg/flare but
holds all the logic to create and send flares.

The API for the component has been cleaned and params are now given at
initialization instead.

The logic to send flare is, for now, in the helpers package to be used
by binaries still relying on pkg/flare.

hush-hush committed:
Initializing flare component through FX

This removes all the duplicate logic about default log files.
We are also now using the flare comp to generate flares from the GUI,
CLI, agent API and systray.

hush-hush committed:
Migrate code out of the flare pkg into comp or the CLI

Only the CLI can generate the pprof profiles. Moving the logic around
them to the CLI and improving testing. This remove dependencies on
pkg/flare and simplify the flare logic.

hush-hush committed:
Use new flare builder in pkg/flare/archive

ogaca-dd reviewed (commented):
LGTM. I didn't spot any issues but it would be nice to have another reviewer as it is very to track changes and see if the new code behaves the same as the old ones.

maxday reviewed (approved):
✅ it seems that it does not touch any serverless-related files

Kaderinho commented on comp/core/flare/helpers/builder.go:
```diff
@@ -109,6 +110,8 @@ type builder struct {
 	flareDir string
 	// permsInfos stores the original rights for the files/dirs that were copied into the flare
 	permsInfos permissionsInfos
+	// isLocal is true when the flare is not generated by the runnning Agent process
```
```suggestion
	// isLocal is true when the flare is not generated by the running Agent process
```

ogaca-dd commented on comp/core/flare/flare.go:
```diff
@@ -6,49 +6,91 @@
+	for _, p := range providers {
+		err = p.Callback(fb)
+		if err != nil {
+			f.log.Error("error calling '%s' for flare creation: %s",
```
f.log.Errorf

sarah-witt reviewed (commented):
Should we use the new `Send` method in the dcaflare https://github.com/DataDog/datadog-agent/blob/main/pkg/cli/subcommands/dcaflare/command.go#L125?
